runelite-client/src/main/java/net/runelite/client/plugins/inferno/InfernoNPC.java

```diff
@@ -25,162 +25,373 @@
-				attackAnimation = 7583;
-				attackstyle = Attackstyle.RANGE;
-				priority = 6;
-				break;
-
-			case NpcID.JALAKREKMEJ:
-				attackTicks = 4;
-				name = "lil mage";
-				attackAnimation = 7581;
-				attackstyle = Attackstyle.MAGE;
-				priority = 5;
-				break;
-
-			case NpcID.JALMEJRAH:
-				attackTicks = 3;
-				name = "bat";
-				attackAnimation = 7578;
-				attackstyle = Attackstyle.RANGE;
-				priority = 4;
-				break;
-
-			case NpcID.JALAK:
-				attackTicks = 6;
-				name = "blob";
-				attackAnimation = 7583; // also 7581
-				attackstyle = Attackstyle.RANDOM;
-				priority = 3;
-				break;
-
-			case NpcID.JALIMKOT:
-				attackTicks = 4;
-				name = "meleer";
-				attackAnimation = 7597;
-				attackstyle = Attackstyle.MELEE;
-				priority = 2;
-				break;
-
-			case NpcID.JALXIL:
-				attackTicks = 4;
-				name = "ranger";
-				attackAnimation = 7605;
-				attackstyle = Attackstyle.RANGE;
-				priority = 1;
-				break;
-
-			case NpcID.JALZEK:
-				attackTicks = 4;
-				name = "mager";
-				attackAnimation = 7610;
-				attackstyle = Attackstyle.MAGE;
-				priority = 0;
-				break;
-
-			default:
-				attackTicks = 0;
+			InfernoNPC.Attack nextBlobAttack = InfernoNPC.Attack.UNKNOWN;
+			if (client.isPrayerActive(Prayer.PROTECT_FROM_MISSILES))
+			{
+				nextBlobAttack = InfernoNPC.Attack.MAGIC;
+			}
+			else if (client.isPrayerActive(Prayer.PROTECT_FROM_MAGIC))
+			{
+				nextBlobAttack = InfernoNPC.Attack.RANGED;
+			}
+
+			this.updateNextAttack(nextBlobAttack);
 		}
+
+		// - This is for jad (jad's animation lasts till after the attack is launched, which fucks up the attack cycle)
+		lastAnimation = this.getNpc().getAnimation();
+		// - This is for blob (to check if player just came out of safespot)
+		lastCanAttack = this.canAttack(client, client.getLocalPlayer().getWorldLocation());
 	}
 
-	public String info()
+	@Getter(AccessLevel.PACKAGE)
+	enum Attack
 	{
-		String info = "";
+		MELEE(Prayer.PROTECT_FROM_MELEE,
+			Color.ORANGE,
+			Color.RED,
+			new int[]{
+				AnimationID.JAL_NIB,
+				AnimationID.JAL_AK_MELEE_ATTACK,
+				AnimationID.JAL_IMKOT,
+				AnimationID.JAL_XIL_MELEE_ATTACK,
+				AnimationID.JAL_ZEK_MELEE_ATTACK, //TODO: Yt-HurKot attack animation
+			}),
+		RANGED(Prayer.PROTECT_FROM_MISSILES,
+			Color.GREEN,
+			new Color(0, 128, 0),
+			new int[]{
+				AnimationID.JAL_MEJRAH,
+				AnimationID.JAL_AK_RANGE_ATTACK,
+				AnimationID.JAL_XIL_RANGE_ATTACK,
+				AnimationID.JALTOK_JAD_RANGE_ATTACK,
+			}),
+		MAGIC(Prayer.PROTECT_FROM_MAGIC,
+			Color.CYAN,
+			Color.BLUE,
+			new int[]{
+				AnimationID.JAL_AK_MAGIC_ATTACK,
+				AnimationID.JAL_ZEK_MAGE_ATTACK,
+				AnimationID.JALTOK_JAD_MAGE_ATTACK
+			}),
+		UNKNOWN(null, Color.WHITE, Color.GRAY, new int[]{});
+
+		private final Prayer prayer;
+		private final Color normalColor;
+		private final Color criticalColor;
+		private final int[] animationIds;
 
-		if (attacking)
+		Attack(Prayer prayer, Color normalColor, Color criticalColor, int animationIds[])
```

Comment:
```suggestion
		Attack(Prayer prayer, Color normalColor, Color criticalColor, int[] animationIds)
```

We try to use java style arrays instead of c-style arrays everywhere